subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -747,8 +745,11 @@ int bt_mesh_scene_srv_set(struct bt_mesh_scene_srv *srv, uint16_t scene,
 
 	BT_DBG("0x%x", scene);
 
-	srv->prev = srv->curr;
-	if (transition && model_transition_is_active(transition)) {
+	if (srv->curr != scene) {
+		srv->prev = srv->curr;
+	}
+	if (transition && model_transition_is_active(transition) &&
+	    (srv->curr != scene)) {
```

Comment:
This need to be there because of 5.3.3.2.2 "If the target state is equal to the current state, the transition shall not be started and is considered
complete."

Reply:
I think there are two ways to read this sentence, and this PR picks a bit of both ways because of this line. 
Either, 
A) the "transition" is referring to the full scene change across all the models, or 
B) it's just referring to the transition of `srv->curr`, ie what's visible in the scene status. 
I believe both are valid interpretations, but we need to pick one and stick with it.

The old version of this function subscribed to the first interpretation: We didn't change anything if the state was the same - not the `srv->curr` state, nor the transition of the other models' state. This PR changes our implementation to the second interpretation: The transition isn't started on the `srv->curr` state, but we'll do the scene change on the other models.

This line is a bit of both though. We're starting the change on the other models, but we're clobbering the transition time to make it *look* like the transition isn't happening.

If we remove this line, the scene_status will report an ongoing transition, which I suppose is the problem this tries to solve. I think we this prevention should happen in `scene_status_encode` instead, which could change its `if (now < srv->transition_end)` to `if (srv->curr != srv->prev)`. This way, we'll do the transition properly on the other models AND follow the rule from the spec, as understood in interpretation B.

Reply:
This has now been rewritten. With a prev and next state and api functions to get current and target state.